app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -128,17 +128,27 @@ def mem_value_eql?(current, desired)
 
   def start_kube_monitor(resource = :pods)
     require 'http'
+    require 'concurrent/atomic/event'
+
+    monitor_started = Concurrent::Event.new
+
     Thread.new do
-      _log.info("Started new #{resource} monitor thread of #{Thread.list.length} total")
+      _log.info("Starting new #{resource} monitor thread of #{Thread.list.length} total")
       begin
-        send(:"monitor_#{resource}")
+        send(:"monitor_#{resource}", monitor_started)
       rescue HTTP::ConnectionError => e
         _log.error("Exiting #{resource} monitor thread due to [#{e.class.name}]: #{e}")
       rescue => e
         _log.error("Exiting #{resource} monitor thread after uncaught error")
         _log.log_backtrace(e)
+      ensure
+        monitor_started.set
       end
     end
+
+    monitor_started.wait
```

Comment:
NOTE it'd be more efficient to start both threads then wait on both Concurrent::Events so they can both run collect_initial at the same time

Reply:
yeah, makes sense to me